examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,212 @@
+
+```bash
+cd docker-image
+export REGION=$(gcloud config get compute/region)
+export PROJET_ID=$(gcloud config get core/project)
+export REPOSITORY=my-repository
+gcloud builds submit --tag ${REGION}-docker.pkg.dev/${PROJECT_ID}/${REPOSITORY}/nvflare-tensorflow:1.0.0
+```
+
+Once the image is built, modify the `kustomization.yaml` file to add the new name and new tag of the image.
+
+## 2. Create persistent volume for Nvflare workspace
+
+All the models generated will be stored in a Cloud storage bucket mounted by each pod. First, create a bucket:
+
+```bash
+gcloud storage buckets create gs://nvflare-storage
+```
+
+Add the following permissions to the Kubernetes Service Account:
+
+```bash
+gcloud storage buckets add-iam-policy-binding gs://nvflare-storage \
+    --member "principal://iam.googleapis.com/projects/391818703174/locations/global/workloadIdentityPools/lgu-demos.svc.id.goog/subject/ns/default/sa/default" \
+    --role "roles/storage.objectUser"
+```
+
+## 3. Create Nvflare folder structure
+
+Now that the persistent volume is created, you can start creating the folder structure and upload it to GCS. First install nvflare on your workstation:
+
+```bash
+python3 -m pip install -r requirements.txt
+```
+
+Then, run the provisionning tool:
+
+```bash
+cd ${HOME}
+nvflare provision
+```
+
+You have to choose the non-ha (non high-availability) choice, as ha (high-availability) is not yet supported on Kubernetes. It will create a `project.yml` file that you can customize for your needs. Then, re-run the provisionning tool:
+
+```bash
+cd ${HOME}
+nvflare provision
+cd ${HOME}/workspace/example_project/prod_00/
+```
+
+The different folders generated represent the infrastructure you will deploy on the reference architecture:
+
+- `server1` is the server that will aggregate all the results from the computation
+- `site-1` and `site-2` are the clients that will be connected to the server
+- `admin@nvidia.com` is the administration client to start and list jobs
+
+## 4. Clone the repository
+
+You will need to clone the Nvflare repository that contains the job you will run on the reference architecture:
+
+```bash
+cd ${HOME}
+git clone https://github.com/NVIDIA/NVFlare
+```
+
+In the newly create folder, there is an `examples` folder that contains lots of demo you can use to test Nvflare. The TensorFlow demo is in the `examples/hello-world/hello-tf2`. The job will need to be in a special folder inside the `admin@nvidia.com` called the `transfer` folder to be deployed on each client participating in the computation.
+Copy this folder in the `transfer` folder of the `admin@nvidia.com` folder:
+
+```bash
+cd ${HOME}/workspace/example_project/prod_00/admin@nvidia.com/transfer
+cp -R ${HOME}/NVFlare/examples/hello-world/hello-tf2 .
+```
+
+Now, copy the whole workspace folder in GCS. The pods will have access to the infrastructure to run jobs:
```

Comment:
GCS -> Cloud Storage.

Let's not use acronyms in this doc, it's not that long. Please, check the rest of the doc as well.

Reply:
Is it better ?